pkg/stanza/fileconsumer/internal/archive/archive.go

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package archive // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/archive"
+
+import (
+	"os"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fileset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fingerprint"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+)
+
+const _ = "knownFiles"
+
+type Archive interface {
+	SetStorageClient(operator.Persister)
+	Match([]*FileRecord) ([]*reader.Reader, error)
+	Write([]*reader.Metadata) error
+}
+
+type FileRecord struct {
+	file *os.File
```

Comment:
What does storing the file handle give us?

Reply:
@djaglowski storing file handle is crucial for us because of the way we handle the archiving:

We try to match files against knownFiles array.
If a match isn't found, we would accumulate the unmatched files (instead of finding in the archive straightaway). More about it was discussed https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32727#issuecomment-2260891596.
Perform the archiving logic. Described [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/431ab4163432fe03658ddd0f34e89598db4abad7/pkg/stanza/fileconsumer/internal/archive/archive.go#L51-L59).
We need the file handle so that we can create a reader from it after the archiving is done.

Reply:
This doesn't explain:
- Why `os.File` is necessary. (And I still don't understand how it's even possible to store and reload it without it being invalidated.)
- Why `fingerprint.Fingerprint` is _sufficient_, when otherwise we use `reader.Metadata` which includes additional information that we deem critical elsewhere such as offset and attributes.